starport/cmd/network_chain_show.go

```diff
@@ -0,0 +1,238 @@
+package starportcmd
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/tendermint/starport/starport/pkg/cosmosutil"
+	"github.com/tendermint/starport/starport/pkg/entrywriter"
+	"github.com/tendermint/starport/starport/pkg/yaml"
+	"github.com/tendermint/starport/starport/services/network"
+	"github.com/tendermint/starport/starport/services/network/networkchain"
+	"github.com/tendermint/starport/starport/services/network/networktypes"
+)
+
+type ShowType string
+
+const (
+	chainShowInfo       ShowType = "info"
+	chainShowGenesis    ShowType = "genesis"
+	chainShowAccounts   ShowType = "accounts"
+	chainShowValidators ShowType = "validators"
+	chainShowPeers      ShowType = "peers"
+)
+
+var (
+	showTypes = map[ShowType]struct{}{
+		chainShowInfo:       {},
+		chainShowGenesis:    {},
+		chainShowAccounts:   {},
+		chainShowValidators: {},
+		chainShowPeers:      {},
+	}
+
+	chainGenesisValSummaryHeader = []string{"Genesis Validator", "Self Delegation", "Peer"}
+	chainGenesisAccSummaryHeader = []string{"Genesis Account", "Coins"}
+	chainVestingAccSummaryHeader = []string{"Vesting Account", "Total Balance", "Vesting", "EndTime"}
+)
+
+// NewNetworkChainShow creates a new chain show command to show
+// a chain on SPN.
+func NewNetworkChainShow() *cobra.Command {
+	c := &cobra.Command{
+		Use:   "show [info|genesis|accounts|validators|peers] [launch-id]",
+		Short: "Show details of a chain",
+		RunE:  networkChainShowHandler,
+		Args:  cobra.ExactArgs(2),
+	}
+
+	c.Flags().AddFlagSet(flagSetKeyringBackend())
+	c.Flags().AddFlagSet(flagNetworkFrom())
+	c.Flags().AddFlagSet(flagSetHome())
+
+	return c
+}
+
+func networkChainShowHandler(cmd *cobra.Command, args []string) error {
+	showType := ShowType(args[0])
+	if _, ok := showTypes[showType]; !ok {
+		cmd.Usage()
+		return fmt.Errorf("invalid arg %s", showType)
+	}
+
+	nb, err := newNetworkBuilder(cmd)
+	if err != nil {
+		return err
+	}
+	defer nb.Cleanup()
+
+	// parse launch ID.
+	launchID, err := network.ParseLaunchID(args[1])
+	if err != nil {
+		return err
+	}
+
+	n, err := nb.Network()
+	if err != nil {
+		return err
+	}
+
+	chainLaunch, err := n.ChainLaunch(cmd.Context(), launchID)
+	if err != nil {
+		return err
+	}
+
+	content := ""
+	switch showType {
+	case chainShowGenesis:
+		content, err = formatChainGenesis(nb, chainLaunch)
+	case chainShowInfo:
+		content, err = formatChainInfo(cmd.Context(), chainLaunch)
+	case chainShowAccounts:
+		content, err = formatChainAccounts(cmd.Context(), n, launchID)
+	case chainShowValidators:
+		content, err = formatChainValidators(cmd.Context(), n, launchID)
+	case chainShowPeers:
+		content, err = formatChainPeers(cmd.Context(), n, launchID)
+	}
+	if err != nil {
+		return err
+	}
+
+	nb.Spinner.Stop()
+	fmt.Print(content)
+	return nil
+}
+
+func formatChainGenesis(nb NetworkBuilder, chainLaunch networktypes.ChainLaunch) (string, error) {
```

Comment:
```suggestion
func formatChainGenesis(out io.Writer, nb NetworkBuilder, chainLaunch networktypes.ChainLaunch) (string, error) {
```

and same for others

Reply:
https://github.com/tendermint/starport/pull/1892#discussion_r768961055